### PR TITLE
Deduplicate JavaScript in frames

### PR DESCRIPTION
**Pull request by srawlins**
The JS in each frame is identical, and I aim to make it longer with changes for Firebase.

### CODE REVIEW

**Comment by srawlins** [comment]
I only spot checked this with `grind serve`. Ought I check modes other than playground, like embedding? Not sure how to manually test that locally.

File: web/scripts/frame.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (srawlins) [inline web/scripts/frame.js:1]: I noticed many of our checked-in JS files do not have a copyright at the top. Should they?

**Review by parlough — APPROVED** [review]

File: web/scripts/frame.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        body.innerHTML = obj.html;
+        document.getElementById('styleId').innerHTML = obj.css;
+        replaceJavaScript(obj.js);
+    }
+};
+
+window.addEventListener('load', function () {
+    window.addEventListener('message', messageHandler, false);
+    parent.postMessage({ 'sender': 'frame', 'type': 'ready' }, '*');
+});
```
Comment (RedBrogdon) [inline web/scripts/frame.js:36]: nit: could use a newline here.

**Review by RedBrogdon — APPROVED** [review]

**Comment by RedBrogdon** [comment]
> I only spot checked this with `grind serve`. Ought I check modes other than playground, like embedding? Not sure how to manually test that locally.

The embedding examples page (https://github.com/dart-lang/dart-pad/blob/master/web/example/embed.html) is always a good resource for this kind of thing. 

I assume this change can't cause some weird race, since the parent should be waiting to hear that "ready" message from the iframe before doing anything. It would be a good thing to try and check, though.

File: web/scripts/frame.js (hunk 1 above)
Comment (srawlins) [reply web/scripts/frame.js:1]: Yeah I'd be happy to. We only have a handful of JS files. @RedBrogdon do you know?

**Review by johnpryan — APPROVED** [review]

File: web/scripts/frame.js (hunk 1 above)
Comment (srawlins) [reply web/scripts/frame.js]: I'll make this change in a follow-up.

File: web/scripts/frame.js (hunk 2 above)
Comment (srawlins) [reply web/scripts/frame.js]: Done

**Comment by srawlins** [comment]
> The embedding examples page (https://github.com/dart-lang/dart-pad/blob/master/web/example/embed.html) is always a good resource for this kind of thing.

Perfect. That's a great little resource. A playground for building our playground.

**Pull request merged**